### PR TITLE
[BWA-16] Import Google Authenticator exports via scanner

### PR DESCRIPTION
**Pull request by SaintPatrck**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/BWA-16

## 📔 Objective

Allow users to import their Google Authenticator tokens.

Users can import by scanning the generated QR code just as if they were scanning a standard OTP auth QR code.

## 📸 Screenshots

authenticator-google-auth-import

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by KatherineInCode — APPROVED** [review]

File: app/src/main/kotlin/com/bitwarden/authenticator/ui/authenticator/feature/itemlisting/ItemListingViewModel.kt
Comment (fedemkr) [inline app/src/main/kotlin/com/bitwarden/authenticator/ui/authenticator/feature/itemlisting/ItemListingViewModel.kt:323]: ❓ Not sure if it's a plausible scenario but, if so, shall we provide a specific error to the user when there are no codes in the payload with TOTP type? So it's easier for them to narrow down what went wrong.

File: app/src/main/kotlin/com/bitwarden/authenticator/data/authenticator/manager/TotpCodeManager.kt
Comment (fedemkr) [inline app/src/main/kotlin/com/bitwarden/authenticator/data/authenticator/manager/TotpCodeManager.kt:38]: ⛏️ IMO, If this is only needed on Google Authenticator, I'd prefix it with `GOOGLE_` or something that indicates that relation so one can guess without the comment what the param is about. Maybe even `GOOGLE_EXPORT_DATA_PARAM` to maintain the same pattern as below in `GOOGLE_EXPORT_PREFIX`.

Moreover, I'd move the Google specific constants to its own object so it's not mixed with the other general ones to keep the general implementation clean and then specific cases can be added elsewhere. So if then we need to add another authenticator's parameters/prefixes, we can do it cleanly in it's own object. Also with a specific object you save the `GOOGLE_` prefix.

**Review by fedemkr — APPROVED** [review]
🎉 Approving with just small comments, but nothing to block the PR.

File: app/src/main/kotlin/com/bitwarden/authenticator/ui/authenticator/feature/itemlisting/ItemListingViewModel.kt
Comment (SaintPatrck) [reply app/src/main/kotlin/com/bitwarden/authenticator/ui/authenticator/feature/itemlisting/ItemListingViewModel.kt:323]: It's totally plausible if a user only enters HOTP codes into Google Authenticator. I like your idea, and easy enough to achieve. I have another PR on deck that adds a Google option to the Import screen. I'll add your suggested notification there.

File: app/src/main/kotlin/com/bitwarden/authenticator/data/authenticator/manager/TotpCodeManager.kt
Comment (SaintPatrck) [reply app/src/main/kotlin/com/bitwarden/authenticator/data/authenticator/manager/TotpCodeManager.kt:38]: Noted, and I concur. We're going to reuse logic here in an upcoming PR, so I'll create a QR code parser abstraction in that one.

**Pull request merged**
